git/05.merge-strategies.md

```diff
@@ -0,0 +1,116 @@
+## [Home](../README.md)
+
+## 5. Merge Strategies
+
+Once your feature branch is approved by a reviewer and ready to merge, how do
+you merge it into the [release branch](07.release-branching.md)?
+
+The [Git Development Workflow](02.development-workflow.md) best practice
+describes the process at a high level, including rebasing, squashing, merging,
+pushing, and cleaning up. This document provides more detail about the merge
+step.
+
+### Background
+
+There are two types of merges: fast-forward (FF) merges and non-fast-forward
+(non-FF) merges. Also, non-FF merges can be _linearizable_, meaning there is no
+ambiguity about the logically linear ordering of commits. Note that linearizable
+non-FF merges can be reduced to a FF merge, a property that we discuss below.
+
+Let's consider the difference with an example. Say this is your starting point
+(ignoring the complexity of a release branch for now):
+
+```
+A   B   C
+o---o---o master
+     \
+      F   G
+      o---o feature
+```
+
+A non-linearizable non-FF merge of `feature` into `master` would look like this:
+
+```
+A   B   C    M
+o---o---o---o master
+     \     /
+      F   G
+      o---o feature
+```
+
+Here `M` is a _merge commit_, which is a commit with more than one parent
+commit. Note that even if the `feature` branch is deleted, the structure of the
+commit history remains the same, and `M` is still needed.
+
+The other two examples require rebasing `feature` onto master first, resulting
```

Comment:
Actually, as I think about it, I think that's the perfect solution: consolidate the code snippets at the end of the document. Then they don't distract from the flow, and are easily skippable or skimmable by those already comfortable with those aspects of git, yet still available for those who aren't. I just added a commit to accomplish this; please let me know what you think. And good idea. :-)